docs/tutorial/7-snapping.md

```diff
@@ -0,0 +1,588 @@
+This can be either a remote Git repository or local files within the project directory.
+Since you want to build the `.c` files that in root of the project directory, you'll set the `source` to `.`:
+
+```yaml
+parts:
+  src:
+    plugin: make
+    source: .
+```
+With this part defined, you can now build the snap:
+
+```{terminal}
+   :input: snapcraft -v
+   :user: ubuntu
+   :host: ubuntu
+   :dir: ~/tutorial
+
+Starting snapcraft, version 8.7.3
+Logging execution to '/home/ubuntu/.local/state/snapcraft/log/snapcraft-20250326-122052.993463.log'
+lxd (5.21/stable) 5.21.3-75def3c from Canonical✓ installed
+Launching managed ubuntu 24.04 instance...
+Creating new instance from remote
+Creating new base instance from remote
+Creating new instance from base instance
+Starting instance
+Starting snapcraft, version 8.7.3
+Logging execution to '/tmp/snapcraft.log'
+Initialising lifecycle
+Installing build-packages
+Installing build-snaps
+Pulling src
+Building src
+:: + make -j12
+:: gcc -Wall -Wextra -Wno-unused-result -O2 -o cfs cfs.c
+:: gcc -Wall -Wextra -Wno-unused-result -O2 -o edf edf.c
+:: gcc -Wall -Wextra -Wno-unused-result -O2 -o fifo fifo.c
+:: gcc -Wall -Wextra -Wno-unused-result -O2 -o lock lock.c
+:: gcc -Wall -Wextra -Wno-unused-result -O2 -o thread-affinity thread-affinity.c
+:: + make -j12 install DESTDIR=/root/parts/src/install
+:: mkdir -p /root/parts/src/install/bin
+:: install -m 755 cfs edf fifo lock thread-affinity /root/parts/src/install/bin
+Staging src
+Priming src
+Packing...
+Reading snap metadata...
+Running linters...
+Running linter: classic
+Running linter: library
+Creating snap package...
+Packed rt-app_0.1_amd64.snap
+```
+
+The first time you build a snap, it may take longer because Snapcraft also downloads and installs a [build provider][build_provider] (as seen in the logs).
+It also provisions an [LXD container][lxd_container].
+
+The build generates an artifact named `rt-app_0.1_amd64.snap`.
+Each part of this name has a meaning:
+- `rt-app`: The snap's name, as defined in the `name` field of :file:`snapcraft.yaml`.
+- `0.1`: The snap's version, as defined in the `version` field of :file:`snapcraft.yaml`.
+  This can be set dynamically based on a git tag or other information, as explained in [this document][using_craftctl_tool].
+- `amd64`: The architecture of the built snap.
+  This matches the host's architecture when doing a native build.
+  The C programs and the snap can also be built for other architectures, for example using the [Canonical build farm][remote_build] or by cross compiling the sources and manually setting the target snap architecture.
+  It is also possible to cross compile using a plugin if the project used [ autotools][cross_compile_autotools].
+
+Now you verify that the real-time applications are pack into the snap, by to inspecting the contents of this `.snap` file.
+Since a snap is a [squashfs] file, as explained in the [snap format document][snap_format], you can unsquash it:
```

Comment:
```suggestion
Since a snap is a [SquashFS file][snap_format] file, you can unsquash it:
```
Hyperlink to wikipedia isn't useful here. Propose adding that hyperlink to the referenced snap format doc.